### PR TITLE
Boolean changes

### PR DESCRIPTION
**Pull request by tomaz1502**

### CODE REVIEW

File: Smt/Reconstruction/Certifying/Benchmark/slow-stats.txt [hunk 1]
```diff
@@ -0,0 +1,218 @@
+direct imports:                        #[Init, Smt.Reconstruction.Certifying.Boolean, Smt.Reconstruction.Certifying.Resolution, Smt.Reconstruction.Certifying.Factor, Smt.Reconstruction.Certifying.PermutateOr]
```
Comment (HanielB) [inline Smt/Reconstruction/Certifying/Benchmark/slow-stats.txt:1]: You should not merge this file.

File: Smt/Reconstruction/Certifying/Examples/tmp.lean [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (HanielB) [inline Smt/Reconstruction/Certifying/Examples/tmp.lean:1]: I assume you'd also not want to merge this one?

File: Smt/Reconstruction/Certifying/Examples/mylean.lean [hunk 3]
```diff
@@ -0,0 +1,166 @@
+
```
Comment (HanielB) [inline Smt/Reconstruction/Certifying/Examples/mylean.lean:1]: If any of these "myLean" examples is to be merged (which would be the case if they are not redundant with other examples), name them in a way that identifies what they are a regression for, like in `badSmtCong` for example.

Comment (tomaz1502) [reply Smt/Reconstruction/Certifying/Examples/mylean.lean:1]: I think for now I won't merge any of these files. Thanks

File: Smt/Reconstruction/Certifying/Util.lean [hunk 4]
```diff
@@ -82,5 +101,16 @@ def getTypeFromName (name : Name) : TacticM Expr :=
 
 def printGoal : TacticM Unit := do
   let currGoal ← getMainGoal
-  let currGoalType ← currGoal.getType
+  let currGoalType ← MVarId.getType currGoal
```
Comment (abdoo8080) [inline Smt/Reconstruction/Certifying/Util.lean:104]: Why the change? I prefer the old current syntax.

File: Smt/Reconstruction/Certifying/Util.lean [hunk 5]
```diff
@@ -82,5 +101,16 @@ def getTypeFromName (name : Name) : TacticM Expr :=
...
+syntax (name := elabTerm) "#elab" term : command
+open Lean.Elab Lean.Elab.Command Lean.Elab.Term in
+def evalElab : CommandElab
+  | `(#elab%$tk $term) => withoutModifyingEnv $ runTermElabM fun _ => do
+    let e ← Term.elabTerm term none
+    unless e.isSyntheticSorry do
+      logInfoAt tk m!"{e} ::: {repr e}"
+  | _ => throwUnsupportedSyntax
+
+/- #elab 2 -/
```
Comment (abdoo8080) [inline Smt/Reconstruction/Certifying/Util.lean:116]:
```suggestion
```

**Review by abdoo8080 — APPROVED** [review]
LGTM! I just have quick comments.
